lib/component-utils/paramsModule.js

```diff
@@ -0,0 +1,11 @@
+function updateParams(oldVnode, vnode) {
+  const elm = vnode.elm;
+  const oldParams = oldVnode.data.params;
+  const params = vnode.data.params;
+  if (!oldParams && !params) return;
+  if (oldParams === params) return;
```

Comment:
This module shouldn't have these if-checks. All null-checks and equality checks should be in the `_setParams` method. Any comparisons between the old & new `vnode.data.params` should be irrelevant to Panel params, the Panel Component should have sole responsibility for when and how new params are applied.